### DISCUSSION TITLE
React Hook Form Wizard, Go back a step

### DISCUSSION THREAD

**Discussion by nickcanarelli** [react-hook-form/react-hook-form#2906] [Q&A]
I am trying to implement a 'go back' a step button but for whatever reason it keeps returning a 404 page not found error when doing so. It appears to only work when going to step 2 and going back to the first step (or default route)..

**Route Setup:**
```
<Router>
    <Route exact path='/onboard/' component={Main} />
    <Route path='/onboard/step2' component={Step2} />
    <Route path='/onboard/step3' component={Step3} />
    <Route path='/onboard/summary' component={Summary} />
</Router>
```

**Go Back function:**
```
const handleBack = () => {
    props.history.go(-1);
};
```

Also, being that it's a nested url, is it necessary to add the following to my onSubmit for Step2 or is there an alternative way of doing so?

```
const onSubmit = (data) => {
    action(data);
    props.history.push('./onboard/step2');
};
```

Everything following the second step seems to work with:
```
const onSubmit = (data) => {
    action(data);
    props.history.push('./summary');
};
```

**Comment by bluebill1049** [comment]
This is probably a router related issue. instead of history.go -1 try to keep pages in an array and navigate between them.
